test/integration/mpi_job_controller_test.go

```diff
@@ -822,6 +823,96 @@ func TestMPIJobWithVolcanoScheduler(t *testing.T) {
 	}
 }
 
+func TestMPIJobManagedExternally(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	s := newTestSetup(ctx, t)
+	startController(ctx, s.kClient, s.mpiClient, nil)
+
+	mpiJob := &kubeflow.MPIJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "job",
+			Namespace: s.namespace,
+		},
+		Spec: kubeflow.MPIJobSpec{
+			SlotsPerWorker: ptr.To[int32](1),
+			RunPolicy: kubeflow.RunPolicy{
+				CleanPodPolicy: ptr.To(kubeflow.CleanPodPolicyRunning),
+				ManagedBy:      ptr.To(kubeflow.MultiKueueController),
+			},
+			MPIReplicaSpecs: map[kubeflow.MPIReplicaType]*kubeflow.ReplicaSpec{
+				kubeflow.MPIReplicaTypeLauncher: {
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "main",
+									Image: "mpi-image",
+								},
+							},
+						},
+					},
+				},
+				kubeflow.MPIReplicaTypeWorker: {
+					Replicas: ptr.To[int32](2),
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "main",
+									Image: "mpi-image",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// 1. The job must be created
+	var err error
+	mpiJob, err = s.mpiClient.KubeflowV2beta1().MPIJobs(s.namespace).Create(ctx, mpiJob, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed sending job to apiserver: %v", err)
+	}
+	backoff := wait.Backoff{
+		Duration: moderateTimeout,
+		Factor:   1.0,
+		Steps:    3,
+	}
+	err = wait.ExponentialBackoff(backoff, func() (bool, error) {
```

Comment:
What is the purpose of using `ExponentialBackoff` here? Can we just wait 1s before doing the asserts? I assume this will be enough to fail if ManagedBy wasn't suppored.

Reply:
It's not very common to use plain sleep in tests, but sure it does the job

Reply:
By using `ExponentialBackoff` you do the sleep anyway, just hidden behind the scenes, and the reason is not clear. 

I think it is better to be transparent and comment on why we sleep, or name the var in a self-explanatory way. 

This is the approach we take in the core k8s: https://github.com/kubernetes/kubernetes/blob/0b91def6bcbdde82206ec779791762fb6bc337af/test/integration/job/job_test.go#L2108.